In [213]:
# # Auxiliares ao desenvolvimento, remover uma vez que o código está funcional
%load_ext autoreload
%autoreload 2
%reload_ext autoreload

<unknown>:14: SyntaxWarning: invalid escape sequence '\O'
<unknown>:16: SyntaxWarning: invalid escape sequence '\O'
<unknown>:18: SyntaxWarning: invalid escape sequence '\O'
<unknown>:24: SyntaxWarning: invalid escape sequence '\c'
<unknown>:30: SyntaxWarning: invalid escape sequence '\%'
<unknown>:32: SyntaxWarning: invalid escape sequence '\c'
<unknown>:14: SyntaxWarning: invalid escape sequence '\O'
<unknown>:16: SyntaxWarning: invalid escape sequence '\O'
<unknown>:18: SyntaxWarning: invalid escape sequence '\O'
<unknown>:24: SyntaxWarning: invalid escape sequence '\c'
<unknown>:30: SyntaxWarning: invalid escape sequence '\%'
<unknown>:32: SyntaxWarning: invalid escape sequence '\c'
<unknown>:14: SyntaxWarning: invalid escape sequence '\O'
<unknown>:16: SyntaxWarning: invalid escape sequence '\O'
<unknown>:18: SyntaxWarning: invalid escape sequence '\O'
<unknown>:24: SyntaxWarning: invalid escape sequence '\c'
<unknown>:30: SyntaxWarning: invalid escape sequence '\%'
<unknown>:32: 

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


<unknown>:14: SyntaxWarning: invalid escape sequence '\O'
<unknown>:16: SyntaxWarning: invalid escape sequence '\O'
<unknown>:18: SyntaxWarning: invalid escape sequence '\O'
<unknown>:24: SyntaxWarning: invalid escape sequence '\c'
<unknown>:30: SyntaxWarning: invalid escape sequence '\%'
<unknown>:32: SyntaxWarning: invalid escape sequence '\c'
<unknown>:14: SyntaxWarning: invalid escape sequence '\O'
<unknown>:16: SyntaxWarning: invalid escape sequence '\O'
<unknown>:18: SyntaxWarning: invalid escape sequence '\O'
<unknown>:24: SyntaxWarning: invalid escape sequence '\c'
<unknown>:30: SyntaxWarning: invalid escape sequence '\%'
<unknown>:32: SyntaxWarning: invalid escape sequence '\c'
<unknown>:14: SyntaxWarning: invalid escape sequence '\O'
<unknown>:16: SyntaxWarning: invalid escape sequence '\O'
<unknown>:18: SyntaxWarning: invalid escape sequence '\O'
<unknown>:24: SyntaxWarning: invalid escape sequence '\c'
<unknown>:30: SyntaxWarning: invalid escape sequence '\%'
<unknown>:32: 

In [214]:
import os
import pandas as pd
from maquinas.dados import Dados
from maquinas.graficos import Graficos
from maquinas.motor_cc_shunt import MotorCCShunt

dados = Dados.ler_dados(nome_arquivo='../dados/dados.txt', separador=",")
motor = MotorCCShunt(dados=dados)
graficos = Graficos(caminho_figuras='../figs/')


Objeto MotorCCShunt criado...


<unknown>:14: SyntaxWarning: invalid escape sequence '\O'
<unknown>:16: SyntaxWarning: invalid escape sequence '\O'
<unknown>:18: SyntaxWarning: invalid escape sequence '\O'
<unknown>:24: SyntaxWarning: invalid escape sequence '\c'
<unknown>:30: SyntaxWarning: invalid escape sequence '\%'
<unknown>:32: SyntaxWarning: invalid escape sequence '\c'
<unknown>:14: SyntaxWarning: invalid escape sequence '\O'
<unknown>:16: SyntaxWarning: invalid escape sequence '\O'
<unknown>:18: SyntaxWarning: invalid escape sequence '\O'
<unknown>:24: SyntaxWarning: invalid escape sequence '\c'
<unknown>:30: SyntaxWarning: invalid escape sequence '\%'
<unknown>:32: SyntaxWarning: invalid escape sequence '\c'


In [215]:
motor.mostrar_resultados()

========================= Parâmetros =========================
Uan   =                240.0
Ian   =                 10.0
Ra    =                  2.0
Rf    =                120.0
Rreos =                    0
Ke    =                  0.8
Kt    =                  0.8
J     =                 0.02
RPM   =               1800.0
Pn    =                  2.5
n     =                 85.0
T_L   =                    0
========================= Resultados =========================
fem                  = 220.0000
w_rads               = 188.4956
torque               = 11.6714
fluxo_tensao         = 1.4589
fluxo_torque         = 1.4589
const_magnetica      = 0.7295
corrente_campo       = 2.0000
corrente_armadura    = 10.0000
velocidade           = 188.4956
potencia_mecânica    = 2200.0000
potencia_elétrica    = 2880.0000
rendimento           = 76.3889


In [220]:
motor_2 = MotorCCShunt(dados=dados)
motor_2.ia = 10.2549

motor_2.mostrar_resultados()

Objeto MotorCCShunt criado...
========================= Parâmetros =========================
Uan   =                240.0
Ian   =              10.2549
Ra    =                  2.0
Rf    =                120.0
Rreos =                    0
Ke    =                  0.8
Kt    =                  0.8
J     =                 0.02
RPM   =               1800.0
Pn    =                  2.5
n     =                 85.0
T_L   =                    0
========================= Resultados =========================
fem                  = 219.4902
w_rads               = 188.4956
torque               = 0.0133
fluxo_tensao         = 1.4555
fluxo_torque         = 0.0016
const_magnetica      = 0.7278
corrente_campo       = 2.0000
corrente_armadura    = 0.0114
velocidade           = 188.4956
potencia_mecânica    = 2.5000
potencia_elétrica    = 2941.1760
rendimento           = 0.0850


# Fluxo de trabalho geral

# Fluxo de trabalho detalhado

1. Criar estruturas para leitura e escrita de arquivos
   1. Essas estruturas devem ler os dados de forma padronizada e retornar um dicionário para posterior uso
      1. O dicionário pode conter campos para comentário
2. Criar uma classe que consuma os dados lidos dos arquivos
3. 

# Condições de operação

## Condição 1: Operação nominal (240 V, sem reostato, carga nominal);




## Condição 2: Subtensão (200 V, carga nominal);




## Condição 3: Enfraquecimento de campo (reostato = 50 Ω);



## Condição 4: Baixa carga (torque reduzido);




## Condição 5: Condição crítica (baixa tensão e alto torque).

In [219]:
#